Bug: reinhardt-commands type-inference regression in 0.1.0-rc.10 without db-sqlite feature

## Prerequisites

- [x] I have searched existing issues to verify this bug hasn't been reported

## Bug Description

`reinhardt-commands` 0.1.0-rc.10 fails to compile when the `db-sqlite` feature is not enabled. The error occurs in `builtin.rs:135` where a `let connection = if ...` expression cannot infer its type because the SQLite branch is conditionally compiled out, leaving insufficient type information.

This is a regression from the fix for #2363 (reinhardt-commands unconditionally references `connect_sqlite` without feature gate). The original issue was fixed by adding `#[cfg(feature = "...")]` guards, but the conditional compilation now causes a type-inference failure when only `db-postgres` (without `db-sqlite`) is enabled.

## Steps to Reproduce

```toml
# Cargo.toml
[dependencies]
reinhardt = { package = "reinhardt-web", version = "0.1.0-rc.10", default-features = false, features = ["database", "db-postgres", "auth", "core", "conf", "server", "commands"] }
```

```bash
cargo check
```

## Expected Behavior

The crate should compile successfully with only `db-postgres` enabled (without `db-sqlite`).

## Actual Behavior

```
error[E0282]: type annotations needed
   --> reinhardt-commands-0.1.0-rc.10/src/builtin.rs:135:21
    |
135 |               let connection = if database_url.starts_with("postgres://")
    |  ______________________________^
...
160 | |             }
    | |_____________^ cannot infer type
```

## Workaround

Include `db-sqlite` in the feature list even if only PostgreSQL is used:

```toml
reinhardt = { package = "reinhardt-web", version = "0.1.0-rc.10", default-features = false, features = ["database", "db-postgres", "db-sqlite", "commands"] }
```

## Environment

- reinhardt-web: 0.1.0-rc.10
- Rust edition: 2024
- rustc: nightly

## Analysis

The `builtin.rs` code at line 135 builds a `connection` value by matching on the database URL prefix. When `db-sqlite` is not enabled, the `else` branch for SQLite is removed by `#[cfg]`, leaving the compiler unable to determine the return type of the if-else chain. A possible fix would be to add an explicit type annotation or an unconditional fallback error branch.

Relates to: #2363

---

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/crates/reinhardt-commands/src/builtin.rs
+++ b/crates/reinhardt-commands/src/builtin.rs
@@ -132,7 +132,7 @@ impl BaseCommand for MigrateCommand {
 			// 4. Connect to database (auto-create if it doesn't exist for PostgreSQL)
 			// This is done before filtering migrations to ensure connection errors are detected
 			// even when no migrations need to be applied
-			let connection = if database_url.starts_with("postgres://")
+			let connection: DatabaseConnection = if database_url.starts_with("postgres://")
 				|| database_url.starts_with("postgresql://")
 			{
 				#[cfg(feature = "postgres")]

